configure.ac

```
@@ -950,6 +950,13 @@ AS_IF([! $arch64],
   [AS_CASE([$ocaml_cv_cc_vendor],
     [gcc-*], [cclibs="$cclibs -latomic"])])
 
+# Support for C11 atomic types
+
+OCAML_CC_SUPPORTS_ATOMIC([$cclibs])
+AS_IF([$cc_supports_atomic],
+  [],
+  [AC_MSG_FAILURE([C11 atomic support is required, use another C compiler])])
```

```suggestion
  [AC_MSG_ERROR([C11 atomic support is required, use another C compiler])])
```
`AC_MSG_FAILURE` adds extra messages to go to `config.log` which I don't think is wanted here (we use it in one other place for unexpected output).

---

Well, the autoconf doc recommends using `AC_MSG_FAILURE` in conjunction with `AC_LINK_IFELSE` to indicate that `config.log` has more explanations for the failure.  I'm just following the advice.

---

It's purely whether we want this style:
```
configure: error: This version of GCC is too old. Please use GCC version 4.9 or above.
```
vs the more verbose:
```
checking whether the C compiler supports _Atomic types... no
configure: error: in `/home/opam/ocaml':
configure: error: C11 atomic support is required, use another C compiler
See `config.log' for more details
```
I guess it depends on one's interpretation of "unexpected" - my only reasoning was that if this test fails, the error message is almost certainly correct (use another C compiler!) and config.log is just going to tell you that the `_Atomic` keyword wasn't recognised!

---

(it can also go in as-is - it definitely works!)

---

Re: "the error message is almost certainly correct", the test can also fail if `-latomic` should have been added but was not, or the "atomic" library wasn't installed.  In all these cases I'd like users to report issues containing the relevant lines of config.log, it will be easier to help them.

---

Ok - let’s go with it! (I’m neutral on the subject of a Changes entry)